tilequeue/command.py

```diff
@@ -948,6 +949,65 @@ def tilequeue_enqueue_tiles_of_interest(cfg, peripherals):
     logger.info('%d tiles of interest processed' % n_toi)
 
 
+def tilequeue_prune_tiles_of_interest(cfg, peripherals):
+    logger = make_logger(cfg, 'prune_tiles_of_interest')
+    logger.info('Pruning tiles of interest')
+
+    logger.info('Fetching tiles of interest ...')
+    tiles_of_interest = peripherals.redis_cache_index.fetch_tiles_of_interest()
+    n_toi = len(tiles_of_interest)
+    logger.info('Fetching tiles of interest ... done. %s found', n_toi)
+
+    logger.info('Fetching tiles recently requested ...')
+    import psycopg2
+
+    redshift_uri = cfg.yml.get('redshift_uri')
+    assert redshift_uri, ("A redshift connection URI must "
+                          "be present in the config yaml")
+
+    tiles_recently_requested = set()
+    with psycopg2.connect(redshift_uri) as conn:
+        with conn.cursor() as cur:
+            cur.execute("""
+                select x, y, z
+                from tile_traffic_v4
+                where (date >= dateadd(day, -30, current_date))
+                  and (z between 0 and 16)
+                  and (x between 0 and pow(2,z)-1)
+                  and (y between 0 and pow(2,z)-1)
+                group by z, x, y
+                order by z, x, y
+            );""")
+            n_trr = cur.rowcount
+            for (x, y, z) in cur:
+                coord = create_coord(x, y, z)
+                coord_int = coord_marshall_int(coord)
+                tiles_recently_requested.add(coord_int)
+
+    logger.info('Fetching tiles recently requested ... done. %s found', n_trr)
+
+    logger.info('Computing tiles of interest to remove ...')
+    toi_to_remove = tiles_of_interest - tiles_recently_requested
+    logger.info('Computing tiles of interest to remove ... done. %s found',
+                len(toi_to_remove))
+
+    logger.info('Removing tiles from TOI and S3 ...')
+
+    def delete_tile_of_interest(coord_int):
+        # Remove from the redis toi set
+        peripherals.redis_cache_index.remove_tile_of_interest(coord_int)
+
+        # Remove the tile from S3
+
+        pass
+
+    for coord_int in toi_to_remove:
+        # FIXME: Think about doing this in a thread/process pool
+        delete_tile_of_interest(coord_int)
```

Comment:
Not sure if this is a [comprehensive list](http://docs.aws.amazon.com/lambda/latest/dg/use-cases.html) of event sources for lambda, but thinking about it more I think that's a reasonable option. One idea is that we can split up the list into groups of 10k or so, push those groups to a location on s3, and have lambda listen to that. Lambda would perform the delete and remove that object from s3.